src/crimson/osd/pg.cc

```diff
@@ -786,11 +801,13 @@ PG::do_osd_ops(
   do_osd_ops_success_func_t success_func,
   do_osd_ops_failure_func_t failure_func)
 {
-  return seastar::do_with(std::move(msg_params), [=, &ops, &op_info]
+  //internal client request
+  SnapContext snapc;  //empty snapc
```

Comment:
Don't capture stack variables by reference unnecessarily.  Remove this declaration and modify below to:

```
        Ref<PG>{this}, std::move(obc), op_info, msg_params, SnapContext{}),
```

Also, reformat to:

```
  return seastar::do_with(
    std::move(msg_params),
    [=, &ops, &op_info, &snapc](auto &msg_params) {
      return do_osd_ops_execute<void>(
        seastar::make_lw_shared<OpsExecuter>(
          Ref<PG>{this}, std::move(obc), op_info, msg_params, SnapContext{}),
        ops,
        std::move(success_func),
        std::move(failure_func));
    });
```